File: src/main/java/de/qabel/core/http/HTTPResult.java
package de.qabel.core.http;

import java.util.Date;

public class HTTPResult<T> {
    int responseCode;
    boolean ok;
    Date lastModified;
    T data;

    public boolean isOk() {
        return ok;
    }

    public void setOk(boolean ok) {
        this.ok = ok;
    }

    public int getResponseCode() {
        return responseCode;
    }

    public void setResponseCode(int responseCode) {
        this.responseCode = responseCode;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public void setLastModified(Date lastModified) {
        this.lastModified = lastModified;
    }

    public Date lastModified() {
        return lastModified;
    }
}

File: core/src/main/java/de/qabel/core/http/HTTPResult.kt
package de.qabel.core.http

import java.util.Date

class HTTPResult<T> {
    var responseCode: Int = 0
    var isOk: Boolean = false
    internal var lastModified: Date? = null
    var data: T? = null

    fun setLastModified(lastModified: Date) {
        this.lastModified = lastModified
    }

    fun lastModified(): Date {
        return lastModified ?: throw IllegalStateException("no last modified date set")
    }
}
